src/cobra/core/metadata/history.py

```diff
@@ -0,0 +1,332 @@
+        self._created_date = HistoryDatetime(date)
+
+    @property
+    def modified_dates(self) -> List:
+        return self._modified_dates
+
+    @modified_dates.setter
+    def modified_dates(self, dates: Iterable[Union[str, "HistoryDateTime"]]) -> None:
+        self._modified_dates = [HistoryDatetime(d) for d in dates]
+
+    @staticmethod
+    def from_data(data: Union[Dict, "History"]) -> "History":
+        """Parse history from data."""
+        if data is None:
+            return History()
+        elif isinstance(data, History):
+            return data
+        elif isinstance(data, dict):
+            return History(**data)
+        else:
+            raise TypeError(f"Unsupported type for History: '{data}'")
+
+    def is_empty(self) -> bool:
+        """Checks if history is empty.
+
+        Returns
+        -------
+        bool
+            Returns False if at least one history attribute is set, else True.
+        """
+        if self.creators:
+            return False
+        if self.created_date.datetime:
+            return False
+        if self.modified_dates:
+            return False
+        return True
+
+    def __eq__(self, history: "History") -> bool:
+        """Checking equality of two history objects.
+
+        A history is equal if all attributes are equal.
+
+        Returns
+        -------
+        bool - True if equal, False otherwise.
+        """
+        # check equality of creators
+        if len(self.creators) != len(history.creators):
+            return False
+        for k, creator in enumerate(self.creators):
+            if creator != history.creators[k]:
+                return False
+
+        # checking equality of created_date
+        if self.created_date != history.created_date:
+            return False
+
+        # checking equality of modified_dates
+        if len(self.modified_dates) != len(history.modified_dates):
+            return False
+        for k, modified_date in enumerate(self.modified_dates):
+            if modified_date != history.modified_dates[k]:
+                return False
+
+        return True
+
+    def to_dict(self) -> Dict:
+        """Returns dictionary representation.
+
+        Returns
+        -------
+        dict - Dictionary representation, of this format
+        {
+        "creators": list[dict]
+        "created_date": str
+        "modified_dates": list[str]
+        """
+        return {
+            "creators": [c.to_dict() for c in self.creators],
+            "created_date": self.created_date.datetime,
+            "modified_dates": [mod_date.datetime for mod_date in self._modified_dates],
+        }
+
+    def __str__(self) -> str:
+        return str(self.to_dict())
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} {self.creators}" \
+               f" {self.created_date.datetime}>"
+
+
+class Creator:
+    """Metadata for person who created an object.
+
+    Parameters
+    ----------
+        given_name : str,
```

Comment:
Creator is defined in the standard.
The standard defines given name, and family name. I generally agree with this, but if there was one name field, it would not be exported/imported well with libsbml. If libsbml will be rewritten, I don't mind changing it.

Reply:
I was wrong. Creator has the option of being with "NAME" or with both "GIVEN NAME" and "FAMILY NAME"
The problem is that some models have both family and given. I could set cobra to read them both into name, and that the cobra class only uses name. Would that be better?